### DISCUSSION TITLE
Add the entire clipboard text to the node past rule finder, possibly as a second argument.

### DISCUSSION THREAD

**Discussion by am-elkheir** [ueberdosis/tiptap#6133] [Feature Requests]
### What problem are you facing?

First of all, thank you for all this magic.
In some cases (also happens to be mine :)), the paste rule regex may allow for new lines. Currently the match will fail on multiple line clipboards because the regex is executing per individual line, as my understanding 😅. 

This is an example of such regex that tries to capture display LaTeX code.

```javascript
/*
clipboard text:

$$
x+y=z
$$

*/

const displayRegex = /\$\$(.)*\$\$/gs


// ...
find: (text) => {
  console.log(text); // '$$', 'x+y=z', then '$$'
  const match = displayRegex .exec(text); // match fails
  // ...
}

```

### What’s the solution you would like to see?

pass the entire clipboard to the finder as a second argument

### What alternatives did you consider?

Maybe there's already a way around it that i am not aware of 😅

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by bdbch** [comment]
I'm not sure about a specific solution that would exist already, but it would be good if pasteHandlers could manage multiple lines too. I don't know when we could work on this feature so I'll keep it open for planning for now. Thanks for your idea!
